### PR TITLE
Flink SQL Engine - GetColumns Operation

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
#1322
#2129

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cxzl25** [comment]
## Fix

image

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+              .filter { tableName => tableNameRegex.pattern.matcher(tableName).matches() }
+              .map { tableName =>
+                val objPath = ObjectIdentifier.of(catalogName, schemaName, tableName).toObjectPath
+                Try(flinkCatalog.getTable(objPath)) match {
+                  case Success(flinkTable) => (tableName, Some(flinkTable))
+                  case Failure(_) => (tableName, None)
+                }
+              }
+              .filter { _._2.isDefined }
+              .flatMap { case (tableName, flinkTable) =>
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala:76]: Is this bunch of code duplicated?

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      pos: Int): Row = {
+    val logicalType = column.getDataType.getLogicalType
+    // format: off
+    Row.of(
+      catalogName,                                                      // TABLE_CAT
+      schemaName,                                                       // TABLE_SCHEM
+      tableName,                                                        // TABLE_NAME
+      column.getName,                                                   // COLUMN_NAME
+      Integer.valueOf(toJavaSQLType(logicalType)),                      // DATA_TYPE
+      logicalType.toString.replace(" NOT NULL", ""), // TYPE_NAME
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala:128]: nit: indent

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      null,                                                             // COLUMN_SIZE
+      null,                                                             // BUFFER_LENGTH
+      getDecimalDigits(logicalType),                                    // DECIMAL_DIGITS
+      getNumPrecRadix(logicalType),                                     // NUM_PREC_RADIX
+      Integer.valueOf(if (logicalType.isNullable) 1 else 0),            // NULLABLE
+      column.getComment.orElse(""),                              // REMARKS
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala:134]: ditto

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    case c: Class[_] if c == classOf[IntType] => 0
+    case c: Class[_] if c == classOf[FloatType] => 7
+    case c: Class[_] if c == classOf[DoubleType] => 15
+    case c: Class[_] if c == classOf[DecimalType] => flinkType.asInstanceOf[DecimalType].getScale
+    case c: Class[_] if c == classOf[TimestampType] => 6
+    case _ => null
+  }
+
+  private def getNumPrecRadix(flinkType: LogicalType): Integer = flinkType.getClass match {
+    case c: Class[_] if c == classOf[TinyIntType] => 10
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetColumns.scala:192]: Could you please merge if branch to one branch?

**Review by SteNicholas — APPROVED** [review]
LGTM. @pan3793 help to merge this pull request.

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Pull request closed without merging**